crates/store/re_types_core/src/arrow_buffer.rs

```diff
@@ -84,56 +83,79 @@ impl<T: bytemuck::Pod> ArrowBuffer<T> {
     }
 }
 
-impl<T: Eq> Eq for ArrowBuffer<T> {}
+impl<T: Eq + ArrowNativeType> Eq for ArrowBuffer<T> {}
 
-impl<T: Clone> ArrowBuffer<T> {
+impl<T: ArrowNativeType> ArrowBuffer<T> {
     #[inline]
     pub fn to_vec(&self) -> Vec<T> {
-        self.0.as_slice().to_vec()
+        self.as_slice().to_vec()
     }
 }
 
-impl<T: arrow::datatypes::ArrowNativeType + arrow2::types::NativeType>
-    From<arrow::buffer::ScalarBuffer<T>> for ArrowBuffer<T>
+impl<T: ArrowNativeType + arrow2::types::NativeType> From<arrow::buffer::ScalarBuffer<T>>
+    for ArrowBuffer<T>
 {
     #[inline]
     fn from(value: arrow::buffer::ScalarBuffer<T>) -> Self {
         Self(value.into_inner().into())
     }
 }
 
-impl<T> From<arrow2::buffer::Buffer<T>> for ArrowBuffer<T> {
+impl<T: ArrowNativeType + arrow2::types::NativeType> From<arrow2::buffer::Buffer<T>>
+    for ArrowBuffer<T>
+{
     #[inline]
-    fn from(value: arrow2::buffer::Buffer<T>) -> Self {
-        Self(value)
+    fn from(arrow2_buffer: arrow2::buffer::Buffer<T>) -> Self {
+        let num_elements = arrow2_buffer.len();
```

Comment:
didn't the fat comment up there say that this is a `num_bytes` in arrow2? That would make this wrong then if I understand correctly

Your unit test seems to pass this though, confusing.

Reply:
This is why I wrote a unit test - arrow2 is confusing and maybe wrongly documented:
* https://github.com/jorgecarleitao/arrow2/issues/1430

The comment is gone too